File: STRIVER/video/vid-7/palindrome.py
def is_palindrome(N):
    n = N
    pal = 0

    if(N<0):
        n = -n

    while(n!=0):
        temp = n%10
        pal = pal*10 + temp
        n = n//10
    return (pal == N)

print(is_palindrome(1234321))

File: STRIVER/video/vid-7 (Maths)/palindrome.cpp
#include <bits/stdc++.h>
using namespace std;

// time complexity  =

bool isPalindrome(int N)
{
    int n = N;
    int pal = 0;

    while (n != 0)
    {
        int temp = n % 10;
        pal = (pal * 10) + temp;
        n = n / 10;
    }

    return (N == pal);
}

int main()
{
    int N = 123;
    cout << isPalindrome(N) << endl;
    return 0;
}